app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -78,17 +106,23 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
             }
     }
 
+    // We need the parameter for the callback
     fun onRandomItem(item: MenuItem) {
+        selectRandomItem()
+    }
+
+    private fun selectRandomItem() {
         lifecycleScope.launch(Dispatchers.IO) {
             val allItems = itemRepository.getAll()
             if (allItems.isNotEmpty()) {
-                val random = allItems.random()
+
+                val random = itemRepository.getAll().random()
                 lifecycleScope.launch(Dispatchers.Main) {
                     val drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
                     drawerLayout.close()
                     getNavController().navigate(
                         ItemsListFragmentDirections
-                            .actionItemsListFragmentToDetailedItemFragment(random)
+                            .actionGlobalDetailedItemFragment(random)
```

Comment:
I am not sure I understand this change. What is the difference between `action...To...` and `actionGlobal...`?

Reply:
The former requires us to be on the items list fragment. This crashes if you're not there, e.g. shaking from somewhere else. Using a global action makes us navigate regardless of where we happen to be.

Reply:
Okay, thanks!